lib/pbench/server/unpack_tarballs.py

```diff
@@ -0,0 +1,135 @@
+import glob
+from logging import Logger
+import os
+from pathlib import Path
+import tempfile
+from typing import NamedTuple
+
+from pbench.server import PbenchServerConfig
+from pbench.server.filetree import FileTree
+from pbench.server.report import Report
+from pbench.server.utils import get_tarball_md5
+
+
+class Results(NamedTuple):
+    total: int
+    success: int
+
+
+class Unpack:
+    LINKSRC = "TO-UNPACK"
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        """Unpacks Tarball in the INCOMING Directory
+
+        Args:
+            config: PbenchServerConfig configuration object
+            logger: A Pbench python Logger
+        """
+        self.config = config
+        self.logger = logger
+
+    def unpack(self, lb_args: int, ub_args: int) -> Results:
+        """Scans for tarballs in the TO-UNPACK subdirectories of the
+        ARCHIVE directory and unpacks them using FileTree.unpack() function.
+
+        Args:
+            lb_args: Lowerbound size of Bucket
+            ub_args: Upperbound size of Bucket
+
+        Returns:
+            Results tuple containing the counts of Total and Successful tarballs.
+        """
+        tarlist = glob.iglob(self.config.ARCHIVE / "*" / self.LINKSRC / "*.tar.xz")
+
+        tarball_list = list()
+        for tarball in tarlist:
+            tar_size = Path(tarball).stat().st_size
+            if tar_size < lb_args or tar_size > ub_args:
+                continue
+            tarball_list.append(tarball)
+
+        ntotal = nsuccess = 0
+
+        file_tree = FileTree(self.config, self.logger)
+
+        for tarball in sorted(tarball_list):
```

Comment:
If you are trying to sort by size (which I think is how the bash script did things), this is not going to work: there is no size information in `tarball_list`, just names, so it is going to sort the names alphabetically.

Reply:
This is a good point.  However, do we really care what order they are processed in?  (Clearly, the answer used to be "yes", but is it still?  I'm hoping that in the future this will all be handled in a totally alien way, so all we have to do here is survive until the future arrives.)

Reply:
I don't think we require "bug-for-bug" compatibility, but we do have several other requirements (although I'm not sure what they _all_ are...):  foremost is to having it implemented in Python; I recommend that it be testable and tested; it needs to be able to unpack tarballs; I don't know whether it has to do them in any particular order...but, if it does do it in order (e.g., alphabetical by name), then it will likely be easier to test reliably.

Reply:
The testability argument for sorting is a good one. I'm happy either way  so I'll resolve this conversation.